pkg/loghttp/query.go

```diff
@@ -42,15 +42,25 @@ const (
 
 // QueryResponse represents the http json response to a Loki range and instant query
 type QueryResponse struct {
-	Status string            `json:"status"`
-	Data   QueryResponseData `json:"data"`
+	Status   string            `json:"status"`
+	Warnings []string          `json:"warnings,omitempty"`
+	Data     QueryResponseData `json:"data"`
 }
 
 func (q *QueryResponse) UnmarshalJSON(data []byte) error {
 	return jsonparser.ObjectEach(data, func(key, value []byte, dataType jsonparser.ValueType, offset int) error {
 		switch string(key) {
 		case "status":
 			q.Status = string(value)
+		case "warnings":
+			var warnings []string
+			if _, err := jsonparser.ArrayEach(value, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+				warnings = append(warnings, string(value))
```

Comment:
Hmm considering JSON can contain all sorts of junk in an array, this probably needs a bit more validation that we are actually getting strings here, presumably through the `dataType` :thinking: 

I assume that `value` doesn't contain the string quote characters, but I wonder if it contains escape characters, e.g. if the warning itself contains a quote symbol, I assume it will be escaped in `value` here?